### PR TITLE
fido2luks: 0.2.20 -> 0.2.21

### LINKED ISSUE #195634: fido2luks: incorrect licensing info
The license listed for `fido2luks` is incorrect. It has been re-licensed upstream: https://github.com/shimunn/fido2luks/commit/4136b1bfad95b996e9575dc454b74b038a227aca

###### Checklist
- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)
- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: `fido2lusk`
current version: `0.2.20`
desired version: n/a

###### Notify maintainers
maintainers: @prusnak @mmahut 

###### Note for maintainers

Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by prusnak**
###### Description of changes

+ new upstream release
+ drop upstreamed patch
+ fix license - fixes https://github.com/NixOS/nixpkgs/issues/195634

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by prusnak** [comment]
@GrahamcOfBorg build fido2luks

**Comment by prusnak** [comment]
@GrahamcOfBorg build fido2luks

**Comment by prusnak** [comment]
@GrahamcOfBorg build fido2luks

File: pkgs/tools/security/fido2luks/default.nix [hunk 1]
```diff
@@ -10,37 +10,28 @@
...
   pname = "fido2luks";
-  version = "0.2.20";
+  version = "0.2.21";
 
   src = fetchFromGitHub {
     owner = "shimunn";
     repo = pname;
     rev = version;
-    sha256 = "04gl7wn38f42mapmkf026rya668vvhm03yi8iqnz31xgggbr2irm";
+    sha256 = "sha256-bXwaFiRHURvS5KtTqIj+3GlGNbEulDgMDP51ZiO1w9o=";
```
Comment (ehllie) [inline pkgs/tools/security/fido2luks/default.nix:19]: I was somewhat under the impression that there was a preference towards using the `hash` attribute over `sha256`. Why not use it here?
```suggestion
    hash = "sha256-bXwaFiRHURvS5KtTqIj+3GlGNbEulDgMDP51ZiO1w9o=";
```

File: pkgs/tools/security/fido2luks/default.nix [hunk 2]
```diff
@@ -10,37 +10,28 @@
...
-
   buildInputs = [ cryptsetup ];
   nativeBuildInputs = [ pkg-config clang ];
 
   configurePhase = ''
     export LIBCLANG_PATH="${llvmPackages.libclang.lib}/lib"
   '';
 
-  cargoSha256 = "U/2dAmFmW6rQvojaKSDdO+/WzajBJmhOZWvzwdiYBm0=";
+  cargoSha256 = "sha256-MPji87jYJjYtDAXO+v/Z4XRtCKo+ftsNvmlBrM9iMTk=";
```
Comment (ehllie) [inline pkgs/tools/security/fido2luks/default.nix:29]: Same as above.
```suggestion
  cargoHash = "sha256-MPji87jYJjYtDAXO+v/Z4XRtCKo+ftsNvmlBrM9iMTk=";
```

**Comment by ehllie** [comment]
Beyond those, it looks good to me

File: pkgs/tools/security/fido2luks/default.nix (hunk 1 above)
Comment (Kranzes) [reply pkgs/tools/security/fido2luks/default.nix:19]: sha256 is still preferred as some programs like `nixpkgs-update` don't know how to handle it.

Comment (ehllie) [reply pkgs/tools/security/fido2luks/default.nix:19]: I rescind my suggestion then

**Comment by prusnak** [comment]
@GrahamcOfBorg build fido2luks

**Pull request merged**